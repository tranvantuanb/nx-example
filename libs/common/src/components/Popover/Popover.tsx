import React from 'react';
import tw, { css, styled } from 'twin.macro';
import { Popover as AntdPopover, PopoverProps as AntdPopoverProps } from 'antd';
import { Global } from '@emotion/react';

export interface PopoverProps extends AntdPopoverProps {
  hideArrow?: boolean;
}

const StyledPopover = styled(AntdPopover)``;

export const Popover: React.FC<PopoverProps> = React.memo(
  ({ hideArrow, ...rest }) => {
    return (
      <>
        <Global
          styles={css`
            .ant-popover-inner-content {
              ${tw`p-[20px]`}
            }
            .hide-arrow {
              .ant-popover-arrow {
                ${tw`hidden`}
              }

              &.ant-popover {
                ${tw`!pt-[5px]`}
              }
            }
          `}
        />
        <StyledPopover
          {...rest}
          overlayClassName={hideArrow ? 'hide-arrow' : ''}
        />
      </>
    );
  }
);
