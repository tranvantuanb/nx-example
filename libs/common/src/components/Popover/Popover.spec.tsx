import { render } from '../../test';

import { Popover, PopoverProps } from './Popover';

describe('Popover', () => {
  it('should render successfully', () => {
    const defaultProps: PopoverProps = {};
    const { container } = render(<Popover {...defaultProps} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
