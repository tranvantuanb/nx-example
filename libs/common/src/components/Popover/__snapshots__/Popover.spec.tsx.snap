// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should render successfully 1`] = `
<div>
  <span
    class="css-1rrze6a-StyledPopover enbff0d0"
  />
</div>
`;
